src/scripts/main.js

```diff
@@ -1,3 +1,20 @@
 'use strict';
 
 // write code here
+const list = document.querySelectorAll('li');
+const listArray = Array.from(list);
+
+const sortedList = listArray.sort((a, b) => {
+  const salaryA = parseInt(a.dataset.salary.replace(/[$,]/g, ''), 10);
+  const salaryB = parseInt(b.dataset.salary.replace(/[$,]/g, ''), 10);
```

Comment:
The conversion of salary strings to numbers should be moved to a separate helper function. Define a function that takes a salary string and returns its numeric value.